### PR TITLE
Ik/some things for text spotting

### PR DESCRIPTION
**Pull request by Ilya-Krylov**

### CODE REVIEW

File: mmdet/core/evaluation/text_evaluation.py [hunk 1]
```diff
@@ -284,8 +287,8 @@ def match(gt_polygons_list, gt_transcriptions, gt_dont_care_polygon_nums,
                         if  match_transcriptions(gt_transcriptions[gt_idx].lower(), pr_transcriptions[pr_idx].lower(), word_spotting):
                             pr_matched_nums.append(pr_idx)
                         else:
-                            print(gt_transcriptions[gt_idx],
-                                  pr_transcriptions[pr_idx])
+                            # print(gt_transcriptions[gt_idx],
+                            #       pr_transcriptions[pr_idx])
```
Comment (druzhkov-paul) [inline mmdet/core/evaluation/text_evaluation.py]: Is it safe to remove the comment as well?

File: mmdet/core/evaluation/text_evaluation.py [hunk 2]
```diff
@@ -43,9 +43,10 @@ def draw_gt_polygons(image, gt_polygons, gt_dont_care_nums):
 
     for point_idx, polygon in enumerate(gt_polygons):
         color = (128, 128, 128) if point_idx in gt_dont_care_nums else (255, 0, 0)
-        for i in range(4):
+        plen = len(polygon[0])
```
Comment (druzhkov-paul) [inline mmdet/core/evaluation/text_evaluation.py:46]: tlen

File: mmdet/datasets/coco_with_text.py [hunk 3]
```diff
@@ -271,14 +282,21 @@ def evaluate(self,
...
             cocoEval.params.imgIds = self.img_ids
-            lexicon_path = metric_params['lexicon']
-            lexicon = self._read_lexicon(lexicon_path)
-
-            lexicon_pairs_path = metric_params['lexicon_mapping']
-            lexicon_mapping = self._read_lexicon_mapping(lexicon_pairs_path)
 
             predictions = []
-            for res in tqdm(results):
+            for img_i, res in tqdm(enumerate(results)):
```
Comment (druzhkov-paul) [inline mmdet/datasets/coco_with_text.py]:
```suggestion
            for img_i, res in enumerate(tqdm(results)):
```

File: mmdet/datasets/coco_with_text.py [hunk 4]
```diff
@@ -271,14 +282,21 @@ def evaluate(self,
...
+                img_id = os.path.basename(cocoGt.imgs[img_i]['filename']).split('.')[0].split('_')[-1]
+
+                lexicon_path = metric_params['lexicon']
+                if lexicon_path and '{}' in lexicon_path:
+                    lexicon_path = lexicon_path.format(img_id)
+                lexicon = self._read_lexicon(lexicon_path)
```
Comment (druzhkov-paul) [inline mmdet/datasets/coco_with_text.py:299]: So, will lexicon be loaded from file on each iteration of the loop even what the lexicon global for all images? Or it is handled effectively somewhere?

File: mmdet/datasets/coco_with_text.py [hunk 5]
```diff
@@ -310,17 +328,47 @@ def evaluate(self,
...
+                            best_recall = recall
+                            best_det_thr = det_thr
+                            best_rec_thr = rec_thr
+
+                eval_results[metric + '/hmean'] = float(f'{best_hmean:.3f}')
+                eval_results[metric + '/precision'] = float(f'{best_precision:.3f}')
+                eval_results[metric + '/recall'] = float(f'{best_recall:.3f}')
+
+                print(f'Text detection recall={best_recall} precision={best_precision} hmean={best_hmean} @ '
+                      f'det_thr={best_det_thr} and rec_thr={best_rec_thr}')
```
Comment (druzhkov-paul) [inline mmdet/datasets/coco_with_text.py:379]: Would it make sense to save thresholds into `eval_results` as well?

File: mmdet/core/evaluation/text_evaluation.py (hunk 2 above)
Comment (Ilya-Krylov) [reply mmdet/core/evaluation/text_evaluation.py:46]: Don't you like the name? Should I change it?

File: mmdet/datasets/coco_with_text.py (hunk 4 above)
Comment (Ilya-Krylov) [reply mmdet/datasets/coco_with_text.py:299]: no, it was not.

Cached.

File: mmdet/datasets/coco_with_text.py (hunk 5 above)
Comment (Ilya-Krylov) [reply mmdet/datasets/coco_with_text.py:379]: fixed

File: mmdet/core/evaluation/text_evaluation.py (hunk 1 above)
Comment (Ilya-Krylov) [reply mmdet/core/evaluation/text_evaluation.py]: removed

File: mmdet/core/evaluation/text_evaluation.py (hunk 2 above)
Comment (druzhkov-paul) [reply mmdet/core/evaluation/text_evaluation.py:46]: No, it's all good.

**Comment by druzhkov-paul** [comment]
run tests

**Review by druzhkov-paul — APPROVED** [review]

**Pull request merged**